src/server/pkg/storage/chunk/transform.go

```diff
@@ -0,0 +1,201 @@
+package chunk
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"crypto/cipher"
+	"encoding/hex"
+	io "io"
+
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+	"github.com/pachyderm/pachyderm/src/server/pkg/obj"
+	"golang.org/x/crypto/chacha20"
+)
+
+type CreateOptions struct {
+	Secret      []byte
+	Compression CompressionAlgo
+}
+
+func Create(ctx context.Context, opts CreateOptions, data []byte, createFunc func(ctx context.Context, data []byte) (ID, error)) (*Ref, error) {
+	buf := make([]byte, len(data))
+	// compression
```

Comment:
I think most of the comments in this file are just filling up space without adding value. The code communicates what is happening clearly enough without them. I think comments are only warranted when it is not obvious what the code is doing. Plus, I think it is better to read the code rather than the comments to understand how something works because it is the source of truth.

Reply:
I removed some of the comments that are just describing what a section of code is doing. But I want to keep the function descriptions.